src/services/uniswap_v3.js

```diff
@@ -0,0 +1,276 @@
+      }
+  }
+
+  async update_pairs(){
+    // Remove banned pairs after ban period
+    if (Object.keys(this.zeroReservePairs).length > 0){
+      for (let pair in this.zeroReservePairs){
+        if (this.zeroReservePairs[pair] <= Date.now()) {
+          delete this.zeroReservePairs[pair];
+          // delete this.tokenList[token];
+        }
+      }
+    }
+    // Generate all possible pair combinations of tokens
+    // This is done by generating an upper triangular matrix or right triangular matrix
+    if (Object.keys(this.tokenSwapList).length > 0){
+      for (let token in this.tokenSwapList){
+        if (this.tokenSwapList[token] <= Date.now()) {
+          delete this.tokenSwapList[token];
+          // delete this.tokenList[token];
+        }
+      }
+
+      let tokens = Object.keys(this.tokenList);
+      var firstToken, secondToken, position;
+      let length = tokens.length;
+      let pairs = [];
+      let pairAddressRequests = [];
+      let pairAddressResponses = [];
+      for (firstToken = 0; firstToken < length; firstToken++){
+        for (secondToken = firstToken + 1; secondToken < length; secondToken++){
+          try{
+            let pairString = this.tokenList[tokens[firstToken]].address + '-' + this.tokenList[tokens[secondToken]].address;
+            if (!this.zeroReservePairs.hasOwnProperty(pairString)){
+              pairs.push(pairString);
+              pairAddressRequests.push(uni.Fetcher.fetchPairData(this.tokenList[tokens[firstToken]], this.tokenList[tokens[secondToken]]));
+            }
+          }
+          catch(err) {
+            logger.error(err);
+          }
+        }
+      }
+
+      await Promise.allSettled(pairAddressRequests).then(values => { for (position = 0; position < pairAddressRequests.length; position++) {
+                                                                      if (values[position].status === "fulfilled"){pairAddressResponses.push(values[position].value)}
+                                                                      else {this.zeroReservePairs[pairs[position]] = Date.now() + this.zeroReserveCheckInterval;}}})
+      this.pairs = pairAddressResponses;
+    }
+    setTimeout(this.update_pairs.bind(this), 1000);
+  }
+
+  async priceSwapIn (tokenIn, tokenOut, tokenInAmount) {
+    await this.extend_update_pairs([tokenIn, tokenOut]);
+    const tIn = this.tokenList[tokenIn];
+    const tOut = this.tokenList[tokenOut];
+    const tokenAmountIn = new uni.TokenAmount(tIn, ethers.utils.parseUnits(tokenInAmount, tIn.decimals));
+    if (this.pairs.length === 0){
+      const route = await this.fetch_route(tIn, tOut);
+      const trade = uni.Trade.exactIn(route, tokenAmountIn);
+      if ( trade !== undefined ){
+        const expectedAmount = trade.minimumAmountOut(this.allowedSlippage);
+        this.cachedRoutes[tIn.symbol + tOut.Symbol] = trade;
+        return { trade, expectedAmount }
+      }
+      return "Can't find route to swap, kindly update "
```

Comment:
Strange this function can return string or object depend on situation. What is it? if this string message is error need throw this message by Error